.github/workflows/tests.yml

```diff
@@ -77,15 +74,17 @@ jobs:
           fi
 
           echo "MIN_COVERAGE=$MIN_COVERAGE" >> "$GITHUB_OUTPUT"
-      - name: Run Unit tests
-        run: poetry run pytest -m "not azure and not functional" --junitxml=coverage-junit.xml --cov=. --cov-report xml:coverage.xml --cov-fail-under ${{ steps.coverage-value.outputs.MIN_COVERAGE }}
-      - uses: actions/upload-artifact@v4
-        if: ${{ !cancelled() }}
+
+      - name: Pre-build image and run unit tests with coverage in dev container
+        uses: devcontainers/ci@v0.3
         with:
-          name: coverage
-          path: |
-            coverage-junit.xml
-            coverage.xml
-          if-no-files-found: error
-      - name: Run Functional tests
-        run: poetry run pytest -m "functional"
+          imageName: ghcr.io/azure-samples/chat-with-your-data-solution-accelerator
+          runCmd: make unittest optional_args="--junitxml=coverage-junit.xml --cov=. --cov-report xml:coverage.xml --cov-fail-under ${{ steps.coverage-value.outputs.MIN_COVERAGE }}"
```

Comment:
Running unit tests separately from others so that we can continue to collect coverage and fail if lower than previous run.